### ISSUE TITLE
Deprecate twisted.persisted.journal

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4298]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)| @thijstriemstra reported|
|-|-|
|Trac ID|trac#4298|
|Type|task|
|Created|2010-02-23 00:46:30Z|
|Branch|https://github.com/twisted/twisted/tree/re-deprecate-persisted-journal-4298|

twisted.persisted.journal is one of the not-quite-reliable storage systems still floating around from Twisted's sordid past.  No one has any intention of finishing it (clearly, as it's been 4 years since its last change), and by current standards we probably wouldn't want it in Twisted anyway due to how far afield from Twisted's primary goals it is.

It should be deprecated so we can remove it in a few more releases.

Attachments:

* [journalDeprecation-4298.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/d01/d015f9e984b26024321334a72e83f3e26b86a73d/cd5e2835df0b7545481b7848c0f8a5f4ba7f753a.diff) (1524 bytes) - added by djfroofy on 2010-02-23 05:06:46Z - 
* [deprecateJournal-4298.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/d01/d015f9e984b26024321334a72e83f3e26b86a73d/4bd2b8debcd8cfeb1d0e2a243b48d6ea8b2f3dd9.diff) (548 bytes) - added by djfroofy on 2010-02-23 05:18:09Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__4298 4298
type__task task
reporter__thijs thijs
priority__normal normal
milestone__ 
branch__branches_re_deprecate_persisted_journal_4298 branches/re-deprecate-persisted-journal-4298
branch_author__cyli__amacleod__djfroofy cyli, amacleod, djfroofy
status__closed closed
resolution__fixed fixed
component__core core
keywords__easy easy
time__1266885990000000 1266885990000000
changetime__1295053312000000 1295053312000000
version__None None
owner__ 
cc__thijs cc__exarkun
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_1"></a>|@itamarst removed owner|
|-|-|

I'd rather not spend typing time on this, but it's certainly worth doing.

**Comment by twisted-trac** [comment]
|[<img alt="djfroofy's avatar" src="https://avatars.githubusercontent.com/u/89760?s=50" width="50" height="50">](https://github.com/djfroofy)<a name="note_3"></a>|@djfroofy commented|
|-|-|

Unit test from first patch is stupid and causes breakage in test suite anyhow.  I'm not sure how to correctly test deprecation warning in module import or if this is even necessary.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_4"></a>|@thijstriemstra commented|
|-|-|

This is blocked by [#4111](https://github.com/twisted/twisted/issues/4111) and doesn't need a higher priority, apps aren't breaking or anything.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set owner to @djfroofy|
|-|-|

Thanks!  Some simple comments:

  1. Missing docstrings on the test class and test method
  1. The test will fail if it is run twice in one process.  You could either:
    1. pop it out of sys.modules before the test runs
    1. deprecate the journal attribute of twisted.persisted instead, using `deprecateModuleAttribute` (although I've never done it this way and I don't know what will happen).

**Comment by twisted-trac** [comment]
|<img alt="amacleod's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_6"></a>|amacleod commented|
|-|-|

(In [28690]) Branching to 'deprecatepersistedjournal-4298'

**Comment by twisted-trac** [comment]
|<img alt="amacleod's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_7"></a>|amacleod commented|
|-|-|

Updated changes in [28590:28694].

**Comment by twisted-trac** [comment]
|<img alt="amacleod's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_8"></a>|amacleod commented|
|-|-|

Er.. I mean [28690:28694].

**Comment by twisted-trac** [comment]
|<img alt="amacleod's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_9"></a>|amacleod commented|
|-|-|

Here's a better revision link: http://twistedmatrix.com/trac/changeset?new=/branches/deprecatepersistedjournal-4298@28694&old=/branches/deprecatepersistedjournal-4298@28690

**Comment by twisted-trac** [comment]
|<img alt="amacleod's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_10"></a>|amacleod commented|
|-|-|

Ready for re-review.

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_11"></a>|@cyli set owner to amacleod|
|-|-|

Deprecation should be "as of 10.1", since the current version of Twisted is 10.0.

Everything else seems to be great though, and the tests pass.  This can probably be merged into trunk after that change.

**Comment by twisted-trac** [comment]
|<img alt="amacleod's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_12"></a>|amacleod commented|
|-|-|

Incremented deprecated-at version in r28714

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_13"></a>|@cyli commented|
|-|-|

Sorry, I should have spotted this last time, but it's missing an entry in the relevant topfile directory listing what change this branch will make (see http://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles).

**Comment by twisted-trac** [comment]
|<img alt="amacleod's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_14"></a>|amacleod set status to `assigned`|
|-|-|

Thanks for the heads-up cyli.  I will add a news file next time I'm home and have free time.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_15"></a>|@thijstriemstra commented|
|-|-|

Thanks. Could you also:

 - update the copyright in twisted/persisted/journal/__init__.py
 - use `2010` instead of `2001-2010` in the new test_journal.py

Merge after that.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_16"></a>|@glyph commented|
|-|-|

Poke, poke, amacleod: this is almost done!  Are you going to take it across the finish line or what? :)

**Comment by twisted-trac** [comment]
|<img alt="amacleod's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_17"></a>|amacleod set status to `closed`|
|-|-|

Finally got off my duff and did it: [29264].

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_18"></a>|@exarkun commented|
|-|-|

Hi amacleod,

Please read http://twistedmatrix.com/trac/wiki/ReviewProcess before your next trunk commit. :)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set status to `reopened`|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="amacleod's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|amacleod removed owner<br>amacleod set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_22"></a>|@exarkun commented|
|-|-|

This was reverted in r29265 because it was missing a news file.  Should be easy for anyone to finish up.

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_23"></a>|@cyli commented|
|-|-|

(In [30451]) Branching to 're-deprecate-persisted-journal-4298'

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_24"></a>|@cyli commented|
|-|-|

(In [30458]) Changed some spacing and whitespace to comply with coding standard.

refs [#4298](https://github.com/twisted/twisted/issues/4298)

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_25"></a>|@cyli commented|
|-|-|

Oops - majority of the changes were in [r30457](http://twistedmatrix.com/trac/changeset/30457)

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_26"></a>|@cyli commented|
|-|-|

Here are some [build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/re-deprecate-persisted-journal-4298)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_28"></a>|@thijstriemstra set owner to @cyli|
|-|-|

Merge after updating the copyright in `twisted/test/test_journal`.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_29"></a>|@thijstriemstra commented|
|-|-|

Replying to [thijs](#4298):
> It should be deprecated so we can remove it in a few more releases.

Opened [#4805](https://github.com/twisted/twisted/issues/4805) for removing it in a future release.

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_30"></a>|@cyli commented|
|-|-|

Replying to [thijs](#note_28):
> Merge after updating the copyright in `twisted/test/test_journal`.

I did so in [r30458](http://twistedmatrix.com/trac/changeset/30458).  It's just when I entered the commit message for [r30457](http://twistedmatrix.com/trac/changeset/30457), I used the word "ref" instead of "refs", so trac didn't pick it up.  Hence I linked to [r30457](http://twistedmatrix.com/trac/changeset/30457) manually, but [r30458](http://twistedmatrix.com/trac/changeset/30458) has some additional changes.  

Anyway, thanks for the review.  Merging.  :)

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_31"></a>|@cyli set status to `closed`|
|-|-|

(In [30460]) Merge re-deprecate-persisted-journal-4298: Deprecate twisted.persisted.journal


Author: djfroofy, amacleod, cyli

Reviewer: thijs, exarkun, cyli

Fixes: [#4298](https://github.com/twisted/twisted/issues/4298)


Re-deprecate twisted.persisted.journal as of 11.0, after reversion (of deprecation as of 10.1) due to lack of newsfile.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
